core/templates/dev/head/components/summary_tile/QuestionSummaryTileDirective.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Summary tile for questions.
+ */
+
+oppia.constant(
+  'QUESTION_EDITOR_URL', '/question_editor/<question_id>');
+
+oppia.directive('questionSummaryTile', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        getQuestionId: '&questionId',
+        getDescription: '&description',
+        getLastUpdatedMsec: '&lastUpdatedMsec',
+        getStatus: '&status',
+        getTaggedSkills: '=taggedSkills'
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/summary_tile/' +
+        'question_summary_tile_directive.html'),
+      controller: [
+        '$scope', 'DateTimeFormatService', 'QUESTION_EDITOR_URL', function(
+            $scope, DateTimeFormatService, QUESTION_EDITOR_URL) {
+          $scope.getLastUpdatedDatetime = function() {
+            return DateTimeFormatService.getLocaleAbbreviatedDatetimeString(
+              $scope.getLastUpdatedMsec());
+          };
+
+          $scope.getQuestionDescription = function () {
+            if ($scope.getDescription()) {
+              // TODO: This description needs to be converted to a readable
+              // and meaningful text.
+              return $scope.getDescription();
+            } else {
+              return 'This question is private. Tag it with a skill and ' +
```

Comment:
If the user viewing the question can publish it, doesn't that mean he/she is the one who created it (or are admins) and therefore can view private questions?

Reply:
Questions appearing on the creator dashboard are the ones who were created by the user, may it be an admin, creator or a TM.

Reply:
Oh, wait, is this text supposed to be placeholder for the question description is it's empty? If so, why isn't this message displayed even if a description does exist? The fact that it is private has nothing to do with it having a description, right?